### DISCUSSION TITLE
redirect not working as i want

### DISCUSSION THREAD

**Discussion by Ebibeck** [vercel/next.js#48357] [Help]
### Summary

cant redirect successfully server flooding client with requests

### Additional information

```js
ceeded. This can happen when a component calls setState inside useEffect, but useEffect either doesn't have a dependency array, or one of the dependencies changes on every render.
```

**Comment by 808vita** [comment]
it would be nice if you could post the code useEffect code block.
and your expected outcome as well.

**Reply by Ebibeck** [reply]
That's the problem it's not a use effect

**Reply by Ebibeck** [reply]
![IMG_20230413_200523_460.jpg](https://user-images.githubusercontent.com/92293938/231895397-95c47bc7-f130-493a-879d-034110d15213.jpg)

![IMG_20230413_200455_854.jpg](https://user-images.githubusercontent.com/92293938/231895398-02f993b3-5a39-4265-a4b6-9e2677defee2.jpg)

I hope you can see it

**Reply by 808vita** [reply]
oh this uses the new app directory ? I have not personally used it yet though .
But I can still vaguely make sense of it .
I think you have created looping expression

```
if (user==null){
redirect("/auth")
}
```
You have placed that in the rootLayout , I'm guessing this wraps the whole app & most probably creating the loop . 
Place additional conditionals like: check if current url path is "/auth" and exit the loop . You could also create an array of paths and check if current path matches (make use of array.includes() ) & then redirect to "/auth".

Or Place the auth check only in specific pages or components .

**Reply by Ebibeck** [reply]
Ok thanks I will try that

**Reply by Ebibeck** [reply]
> oh this uses the new app directory ? I have not personally used it yet though .
> But I can still vaguely make sense of it .
> I think you have created looping expression
> 
> ```
> if (user==null){
> redirect("/auth")
> }
> ```
> You have placed that in the rootLayout , I'm guessing this wraps the whole app & most probably creating the loop . 
> Place additional conditionals like: check if current url path is "/auth" and exit the loop . You could also create an array of paths and check if current path matches (make use of array.includes() ) & then redirect to "/auth".
> 
> Or Place the auth check only in specific pages or components .
> 
> 

Yes it created an infinite loop just flooding my auth page with request

**Reply by 808vita** [reply]
👍👍👍
